synchronizer/synchronizer.go

```diff
@@ -598,7 +598,8 @@ func (s *ClientSynchronizer) processSequenceBatches(sequencedBatches []etherman.
 			if !status {
 				// Reset trusted state
 				log.Infof("reorg detected, discarding batches until batchNum %d", batch.BatchNumber)
-				err := s.state.ResetTrustedState(s.ctx, batch.BatchNumber, dbTx) // This method has to reset the forced batches deleting the batchNumber for higher batchNumbers
+				previousBatchNumber := batch.BatchNumber - 1
+				err := s.state.ResetTrustedState(s.ctx, previousBatchNumber, dbTx) // This method has to reset the forced batches deleting the batchNumber for higher batchNumbers
```

Comment:
```suggestion
				err := s.state.ResetTrustedState(s.ctx, batch.BatchNumber - 1, dbTx) // This method has to reset the forced batches deleting the batchNumber for higher batchNumbers
```

Reply:
@cool-develope the creation of the variable `previousBatchNumber` was intended, this helps the code to express the `-1` was a thought decision instead of something else, this helps with future maintenance and the code health.